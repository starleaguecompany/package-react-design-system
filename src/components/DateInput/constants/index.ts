export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
