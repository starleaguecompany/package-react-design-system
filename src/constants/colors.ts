export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
