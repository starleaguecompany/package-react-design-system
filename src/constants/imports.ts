export const reactDSImportPath = '@starleaguecompany/react-design-system';
