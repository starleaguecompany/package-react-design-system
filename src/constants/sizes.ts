export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
