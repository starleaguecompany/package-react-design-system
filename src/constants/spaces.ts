export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
