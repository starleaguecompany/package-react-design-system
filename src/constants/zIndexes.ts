export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
