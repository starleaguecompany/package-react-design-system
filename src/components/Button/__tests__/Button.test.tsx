import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { CONTAINER_SIZES } from '../../../constants/sizes';
import { Menu } from '@starleaguecompany/react-icons';
import { Button, Icon } from '../../../index';

describe('Button', () => {
  test('should render correctly with no props', () => {
    render(<Button />);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('container size-44 color-gray-outlined', { exact: true });
  });

  test('should render correctly with text', () => {
    render(<Button>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveTextContent('Testing button');
  });

  test('should render correctly with attributes', () => {
    render(
      <Button id="test-id" className="test-class">
        Testing button
      </Button>
    );

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveAttribute('id', 'test-id');
    expect(screen.getByRole('button')).toHaveClass('test-class');
  });

  test('should render correctly with variant prop', () => {
    render(<Button variant="primary">Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('color-gray-primary');
  });

  test('should render correctly with color prop', () => {
    render(<Button color="blue">Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('color-blue-outlined');
  });

  test('should render correctly with color and variant props', () => {
    render(
      <Button color="blue" variant="primary">
        Testing button
      </Button>
    );

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('color-blue-primary');
  });

  test('should render correctly with size prop', () => {
    render(<Button size={CONTAINER_SIZES.S60}>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('size-60');
  });

  test('should render correctly with block prop', () => {
    render(<Button block>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('block');
  });

  test('should render correctly with loading prop', () => {
    render(<Button loading>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('disabled-outlined');
    expect(screen.getByRole('button')).toHaveAttribute('disabled');
    expect(screen.getByRole('button')).toContainElement(screen.getByTestId('Spinner'));
  });

  test('should render correctly with disabled prop', () => {
    render(<Button disabled>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('disabled-outlined');
    expect(screen.getByRole('button')).toHaveAttribute('disabled');
    expect(screen.getByRole('button')).toBeDisabled();
  });

  test('should render correctly with active prop', () => {
    render(<Button active>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveClass('active');
  });

  test('should render correctly with icon', () => {
    render(
      <Button>
        <Icon icon={<Menu />} />
      </Button>
    );

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toContainElement(screen.getByTestId('Menu'));
  });

  test('should execute callBack on click', () => {
    const clickCallback = jest.fn();
    render(<Button onClick={clickCallback}>Testing button</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    userEvent.click(screen.getByRole('button'));
    expect(clickCallback).toBeCalledTimes(1);
  });
});
