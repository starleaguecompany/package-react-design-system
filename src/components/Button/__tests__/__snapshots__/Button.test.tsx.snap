// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should execute callBack on click 1`] = `
<button
  class="container size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with active prop 1`] = `
<button
  class="container active size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with attributes 1`] = `
<button
  class="test-class container size-44 color-gray-outlined"
  data-qa="Button"
  id="test-id"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with block prop 1`] = `
<button
  class="container block size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with color and variant props 1`] = `
<button
  class="container size-44 color-blue-primary"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with color prop 1`] = `
<button
  class="container size-44 color-blue-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with disabled prop 1`] = `
<button
  class="container size-44 disabled-outlined color-gray-outlined"
  data-qa="Button"
  disabled=""
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with icon 1`] = `
<button
  class="container size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    <span
      class="container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </span>
</button>
`;

exports[`Button should render correctly with loading prop 1`] = `
<button
  class="container loading size-44 disabled-outlined color-gray-outlined"
  data-qa="Button"
  disabled=""
>
  <span
    class="text"
  >
    Testing button
  </span>
  <span
    class="spinner container size-20"
    data-qa="Spinner"
  >
    <svg
      class="svg"
      viewBox="0 0 150 150"
    >
      <circle
        class="circle circle-1"
        cx="75"
        cy="75"
        r="60"
      />
      <circle
        class="circle circle-2"
        cx="75"
        cy="75"
        r="60"
      />
    </svg>
  </span>
</button>
`;

exports[`Button should render correctly with no props 1`] = `
<button
  class="container size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  />
</button>
`;

exports[`Button should render correctly with size prop 1`] = `
<button
  class="container size-60 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with text 1`] = `
<button
  class="container size-44 color-gray-outlined"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;

exports[`Button should render correctly with variant prop 1`] = `
<button
  class="container size-44 color-gray-primary"
  data-qa="Button"
>
  <span
    class="text"
  >
    Testing button
  </span>
</button>
`;
